tests/mock_data/kdd/ipam.yaml

```diff
@@ -6,6 +6,8 @@ metadata:
 spec:
   node: kube-master
   cidr: 192.168.221.192/26
+  deleted: ""
```

Comment:
@caseydavenport Am I right that these `deleted: ""` additions should be reverted if your proposed libcalico change (to make `deleted` an option boolean) is merged?

Reply:
@neiljerram Yeah, if we change the type then we'll need to update these lines to `deleted: false`.

The issue is that we were applying yamls here that weren't fully qualified, and the schema mandates that the fields are present. That will be the same even with my PR, independent of the type of the field.

Reply:
Even with the `+optional` tag?

Reply:
Yeah, we could add that and `omitempty` to make it an optional field, and then remove these lines, but I wasn't planning on doing that. It's an internal API and all of the code does actually write these fields out today, so seems like we should keep doing that unless there's a reason.